### PR TITLE
feat: added project data key

### PR DESCRIPTION
**Pull request by sheensantoscapadngan**
# Description 📣
- This PR adds the generation of project data key for secret encryption/decryption

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/services/kms/kms-service.ts [hunk 1]
```diff
@@ -167,35 +167,120 @@ export const kmsServiceFactory = ({
...
-      if (!project) {
-        throw new BadRequestError({ message: "Project not found" });
+    let project = await projectDAL.findById(projectId);
+    if (!project) {
+      throw new BadRequestError({ message: "Project not found" });
+    }
+
+    if (!project.kmsSecretManagerKeyId) {
+      // create default kms key for certificate service
+      const lock = await keyStore
```
Comment (maidul98) [inline backend/src/services/kms/kms-service.ts:177]: We do this so that multiple requests to create the default KMS doesn't create conflict?

Comment (sheensantoscapadngan) [reply backend/src/services/kms/kms-service.ts:177]: yup exactly

Comment (sheensantoscapadngan) [reply backend/src/services/kms/kms-service.ts:177]: last thing we want is for data key to be encrypted by a stale KMS

**Review by maidul98 — APPROVED** [review]
lgtm

**Pull request merged**